refactor iconFrame pools in QuestieFramePool

Laumesis opened the pull request:
Refactored usedFrames, unusedFrames, allFrames
Moved iconFrame (re)init code to outside of GetFrame()
The (re)init code needs to be moved to QuestieFrame.lua

Laumesis committed:
refactor iconFrame pools in QuestieFramePool
Refactored usedFrames, unusedFrames, allFrames
Moved iconFrame (re)init code to outside of GetFrame()
The (re)init code needs to be moved to QuestieFrame.lua

BreakBB commented on Modules/FramePool/QuestieFramePool.lua:
```diff
@@ -16,17 +16,21 @@ local l10n = QuestieLoader:ImportModule("l10n")
 
 local HBDPins = LibStub("HereBeDragonsQuestie-Pins-2.0")
 
+local tinsert = tinsert
+local tremove = tremove
```
```suggestion
local tinsert = table.insert
local tremove = table.remove
```

Laumesis committed:
[fix] increment lineFrameCount only for new frames
use table.insert instead tinsert, even WoW lua provides global tinsert
rename local function to start with _

Laumesis commented:
Added [fix] increment lineFrameCount only for new frames

Still needs The (re)init code needs to be moved to QuestieFrame.lua
